### PR TITLE
Validate parameters

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

**Comment by mdellweg** [comment]
I may be missing some `--*-in` options that are not exercised in tests. So please have a close look if you will.

**Review by ggainey — CHANGES_REQUESTED** [review]
A couple suggestions, but overall looking good. I'd like to see more tests, esp for rpm-content, but that should be its own issue I think.

File: tests/scripts/pulp_rpm/test_content.sh [hunk 1]
```diff
@@ -120,6 +122,10 @@ do
   fi
 done
 
+expect_succ pulp rpm content list --name-in "${RPM_NAME}" --name-in "${RPM2_NAME}"
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply tests/scripts/pulp_rpm/test_content.sh:125]: I have two already...

File: pulpcore/cli/common/generic.py [hunk 2]
```diff
@@ -821,6 +825,10 @@ def callback(
         """
         Show the list of optionally filtered {entities}.
         """
+        if "ordering" in kwargs:
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/cli/common/generic.py:828]: No, even when the user does not specify ordering, we will receive `[]` from click. And that fails validation on GroupSomething and SigningServices.

File: pulpcore/cli/rpm/content.py [hunk 3]
```diff
@@ -136,17 +142,25 @@ def content() -> None:
         allowed_with_contexts=(PulpRpmPackageContext, PulpRpmModulemdContext),
     ),
     pulp_option("--package-href", allowed_with_contexts=(PulpRpmPackageContext,)),
-    pulp_option("--pkgId", allowed_with_contexts=(PulpRpmPackageContext, PulpRpmModulemdContext)),
+    pulp_option("--pkgId", allowed_with_contexts=(PulpRpmPackageContext,)),
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/cli/rpm/content.py:145]: :-D

File: pulpcore/cli/rpm/content.py [hunk 4]
```diff
@@ -111,23 +111,29 @@ def content() -> None:
...
-    pulp_option("--epoch-in", "epoch__in", allowed_with_contexts=(PulpRpmPackageContext,)),
-    pulp_option("--epoch-ne", "epoch__ne", allowed_with_contexts=(PulpRpmPackageContext,)),
-    pulp_option("--id", allowed_with_contexts=(PulpRpmPackageContext, PulpRpmAdvisoryContext)),
     pulp_option(
-        "--id-in", "id__in", allowed_with_contexts=(PulpRpmPackageContext, PulpRpmAdvisoryContext)
+        "--epoch-in", "epoch__in", multiple=True, allowed_with_contexts=(PulpRpmPackageContext,)
     ),
+    pulp_option("--epoch-ne", "epoch__ne", allowed_with_contexts=(PulpRpmPackageContext,)),
+    pulp_option("--id", multiple=True, allowed_with_contexts=(PulpRpmAdvisoryContext,)),
+    pulp_option("--id-in", "id__in", allowed_with_contexts=(PulpRpmAdvisoryContext,)),
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/cli/rpm/content.py:124]: I see now! I accidentally added the multiple to "--id" instead.

File: tests/scripts/pulp_rpm/test_content.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ggainey) [reply tests/scripts/pulp_rpm/test_content.sh:125]: Yeah I have no idea what I was thinking there

**Review by ggainey — APPROVED** [review]

**Pull request merged**